rust/parser/typeql.pest

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+eof_query = { SOI ~ query ~ EOI }
+eof_queries = { SOI ~ query+ ~ EOI }
+eof_pattern = { SOI ~ pattern ~ EOI }
+eof_patterns = { SOI ~ patterns ~ EOI }
+eof_definables = { SOI ~ definables ~ EOI }
+eof_variable = { SOI ~ pattern_variable ~ EOI }
+eof_label = { SOI ~ label ~ EOI }
+eof_schema_rule = { SOI ~ schema_rule ~ EOI }
```

Comment:
SOI = Start of Input, EOI = End of Input, `~` = "followed by". `~` includes optional whitespace and comments.